src/metadata/EmbeddedMetadata.ts

```diff
@@ -32,7 +33,7 @@ export class EmbeddedMetadata {
     /**
      * Embedded target type.
      */
-    type: Function;
+    type: Function | EntitySchema;
```

Comment:
I think what we should do instead is to mark `type` as optional, and don't store anything here if type is entity schema is used.